### DISCUSSION TITLE
GlobalKey<State> alternative?

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#75] [Q&A] [answered]
I want to replicate this in ReArch:

```dart
globalKey.currentState?.requestFocus();
```

Is there a way for getting a `RearchConsumer` by its key?

**Comment by GregoryConrad** [comment, marked as answer]
No, and you probably shouldn't be using `GlobalKey`s for this anyways since they act like global/static variables.

Instead, provide whatever state you have down to the rest of the widget tree via [scoping state](https://rearch.gsconrad.com/flutter/scoping-state).

**Reply by busslina** [reply]
> Instead, provide whatever state you have down to the rest of the widget tree via [scoping state](https://rearch.gsconrad.com/flutter/scoping-state).

So we have to wait to static metaprogramming to use scoping!?

**Reply by GregoryConrad** [reply]
No, you can do it now too, it's just a ton of boilerplate. Make a RearchConsumer that manages some state via side effects, and then make a new InheritedWidget that takes the state of the RearchConsumer in its constructor to provide to descendants.

**Reply by GregoryConrad** [reply]
I recognize that that approach is full of unwanted boilerplate, which is why I plan to make the macro as soon as static metaprogramming comes around

**Reply by GregoryConrad** [reply]
(If you have any questions about making the RearchConsumer + InheritedWidget, please let me know and I can make an example. Probably should make one for the documentation in the meantime anyways)

**Reply by busslina** [reply]
Okay, If I have strength and will, I let you know. Thanks !

**Reply by GregoryConrad** [reply]
FYI, `flutter_rearch` v1.4.0 was released with `use.focusNode` and `use.focusScopeNode` builtin 🎉
